daemons/gptp/common/ieee1588port.cpp

```diff
@@ -573,35 +611,35 @@ void IEEE1588Port::processEvent(Event e)
 				clock->addEventTimerLocked(this, e4, interval4);
 		}
 
-		if (automotive_profile) {
+     	if (clock->getAutomotiveState()) {
 			setStationState(STATION_STATE_ETHERNET_READY);
-			if (testMode) {
+      }
+         	if (clock->getAutomotiveTestMode()) {
 				APMessageTestStatus *testStatusMsg = new APMessageTestStatus(this);
 				if (testStatusMsg) {
 					testStatusMsg->sendPort(this);
 					delete testStatusMsg;
-				}
 			}
-			if (!isGM) {
+      	}
+
+      	if (clock->getExternalPortConfiguration() == EXT_SLAVE
+          	&& clock->getNegotiateSyncRate()) {
 				// Send an initial signalling message
 				PTPMessageSignalling *sigMsg = new PTPMessageSignalling(this);
 				if (sigMsg) {
 					sigMsg->setintervals(PTPMessageSignalling::sigMsgInterval_NoSend, log_mean_sync_interval, PTPMessageSignalling::sigMsgInterval_NoSend);
 					sigMsg->sendPort(this, NULL);
 					delete sigMsg;
 				}
-
 				startSyncReceiptTimer((unsigned long long)
 					 (SYNC_RECEIPT_TIMEOUT_MULTIPLIER *
 					  ((double) pow((double)2, getSyncInterval()) *
 					   1000000000.0)));
-			}
 		}
-
 		break;
 
 	case STATE_CHANGE_EVENT:
-		if (!automotive_profile) {       // BMCA is not active with Automotive Profile
+      	if (clock->getExternalPortConfiguration() == EXT_DISABLED) {       // BMCA is not active with Automotive Profile
```

Comment:
It seems that automotive_profile means a static port configuration? In both Avnu gPTP code and cummings_external_port code, BMCA is disabled when automotive profile is enabled so that the port state is forced and will not change. So I think BMCA should always be turned off when automotive profile is enabled.

Reply:
Hmm okay. I would think making it a feature would be more flexible but it looks like it is technically required in the AVnu automotive profile. I'll defer to Rodney as to whether it's worth the effort right now or not.